primitives/utility/src/lib.rs

```diff
@@ -33,21 +33,37 @@ use xcm::{latest::prelude::*, WrapVersion};
 /// for the `SendXcm` implementation.
 pub struct ParentAsUmp<T, W>(PhantomData<(T, W)>);
 impl<T: UpwardMessageSender, W: WrapVersion> SendXcm for ParentAsUmp<T, W> {
-	fn send_xcm(dest: impl Into<MultiLocation>, msg: Xcm<()>) -> Result<(), SendError> {
-		let dest = dest.into();
+	type Ticket = Vec<u8>;
 
-		if dest.contains_parents_only(1) {
+	fn validate(
+		dest: &mut Option<MultiLocation>,
+		msg: &mut Option<Xcm<()>>,
+	) -> SendResult<Vec<u8>> {
+		let d = dest.take().ok_or(SendError::MissingArgument)?;
+		let xcm = msg.take().ok_or(SendError::MissingArgument)?;
+
+		if d.contains_parents_only(1) {
 			// An upward message for the relay chain.
 			let versioned_xcm =
-				W::wrap_version(&dest, msg).map_err(|()| SendError::DestinationUnsupported)?;
+				W::wrap_version(&d, xcm).map_err(|()| SendError::DestinationUnsupported)?;
 			let data = versioned_xcm.encode();
 
-			T::send_upward_message(data).map_err(|e| SendError::Transport(e.into()))?;
-
-			Ok(())
+			Ok((data, MultiAssets::new()))
```

Comment:
Should make this configurable so that sending UMP messages can have a price.

Reply:
See https://github.com/paritytech/polkadot/pull/4097/files#diff-b560dfbd5905fc793b2ad72e6db42c9f6f3e3d73e234c7076449aa1bd610c2bdR30-R44